wallet-new/docs/updates.md

```diff
@@ -0,0 +1,563 @@
+    |   |                                                |                                                |
+    |   |                   GET /api/v1/update           |                                                |
+    |   +------------------------------------------------>                                                |
+    |   |                                                |                                                |
+    |   |       returns optional update                  |                                                |
+    |   <------------------------------------------------+                                                |
+    |t  |                                                |    mandatory update arrives, data overwritten  |
+    |i  |                                                |                        ------------------------+
+    |m  |                                                <-----------------------/                        |
+    |e  |                                                |                                                |
+    |   |       PUT /api/v1/update/skip                  |                                                |
+    |   +------------------------------------------------> the user just skipped a mandatory update!      |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    |   |                                                |                                                |
+    V   |                                                |                                                |
+
+If we kept only one single field to store this data, such field might be sensible to data races and *ghost updates*,
+where we could accidentally override data in a way which would cause our internal invariants to be violated.
+
```

Comment:
It's unclear when the update goes from "latest" slot to "current". Does it happen when user gets this update with `GET /api/v1/update` at the first time? In which cases update in "latest" slot replaces update in "current" slot and in which doesn't? Does skipped updates always get replaced by latest ones? This all needs clarification.

Reply:
You are right, people cannot read what's on my mind so I should clarify that 😂